dace/frontend/python/replacements.py

```diff
@@ -456,24 +456,50 @@ def _numpy_rot90(pv: 'ProgramVisitor', sdfg: SDFG, state: SDFGState, arr: str, k
 
     k %= 4
 
-    if k == 0:
-        return arr
-    if k == 2:
-        res = _numpy_flip(pv, sdfg, state, arr, axes[0])
-        state = pv._add_state(state.label + '_b')
-        return _numpy_flip(pv, sdfg, state, res, axes[1])
+    to_flip = []
+    transpose = False
 
     axes_list = list(range(ndim))
     (axes_list[axes[0]], axes_list[axes[1]]) = (axes_list[axes[1]], axes_list[axes[0]])
+    inpidx = ','.join([f'__i{i}' for i in range(ndim)])
 
-    if k == 1:
-        res = _numpy_flip(pv, sdfg, state, arr, axes[1])
-        state = pv._add_state(state.label + '_b')
-        return _transpose(pv, sdfg, state, res, axes_list)
+    if k == 0:
+        return arr
+    if k == 2:
+        to_flip = [axes[0], axes[1]]
+    elif k == 1:
+        to_flip = [axes[1]]
+        transpose = True
     else:  # k == 3
-        res = _transpose(pv, sdfg, state, arr, axes_list)
-        state = pv._add_state(state.label + '_b')
-        return _numpy_flip(pv, sdfg, state, res, axes[1])
+        to_flip = [axes[0]]
+        transpose = True
+
+    arr_copy, narr = sdfg.add_temp_transient_like(desc)
+
+    shape_list = list(narr.shape)
```

Comment:
Couldn't we have updated `numpy.flip` instead here? Are the semantics different?

Reply:
The semantics are the same, but we couldn't have updated `flip` because then the maps would not be fused. This was added because the CUDA compiler also had a bug when we tried running the two (flip/transpose) as separate kernels. Fusing them both makes rot90 faster, and works around that bug (which even appears in `-O0` and was reported).